tests/TestCase/Database/CommonTableExpressionIntegrationTest.php

```diff
@@ -1,1027 +0,0 @@
- *
- * Licensed under The MIT License
- * For full copyright and license information, please see the LICENSE.txt
- * Redistributions of files must retain the above copyright notice.
- *
- * @copyright     Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
- * @link          https://cakephp.org CakePHP(tm) Project
- * @since         4.1.0
- * @license       https://opensource.org/licenses/mit-license.php MIT License
- */
-namespace Cake\Test\TestCase\Database;
-
-use Cake\Database\Driver\Mysql;
-use Cake\Database\Driver\Sqlserver;
-use Cake\Database\Expression\CommonTableExpression;
-use Cake\Database\Expression\QueryExpression;
-use Cake\Database\Query;
-use Cake\Database\ValueBinder;
-use Cake\Datasource\ConnectionManager;
-use Cake\TestSuite\TestCase;
-use InvalidArgumentException;
-
-class CommonTableExpressionIntegrationTest extends TestCase
-{
-    /**
-     * @inheritDoc
-     */
-    protected $fixtures = [
-        'core.Articles',
-    ];
-
-    /**
-     * @inheritDoc
-     */
-    public $autoFixtures = false;
-
-    /**
-     * @var \Cake\Database\Connection
-     */
-    protected $connection;
-
-    /**
-     * @var bool
-     */
-    protected $autoQuote;
-
-    public function setUp(): void
-    {
-        parent::setUp();
-        $this->connection = ConnectionManager::get('test');
-
-        $this->skipIf(
-            !$this->connection->getDriver()->supportsCTEs(),
-            'The current driver does not support common table expressions.'
-        );
-
-        $this->autoQuote = $this->connection->getDriver()->isAutoQuotingEnabled();
-    }
-
-    public function tearDown(): void
-    {
-        parent::tearDown();
-        $this->connection->getDriver()->enableAutoQuoting($this->autoQuote);
-        unset($this->connection);
-    }
-
-    public function assertQuotedQuery($pattern, $query, $optional = false): void
-    {
-        if ($optional) {
-            $optional = '?';
-        }
-        $pattern = str_replace('<', '[`"\[]' . $optional, $pattern);
-        $pattern = str_replace('>', '[`"\]]' . $optional, $pattern);
-        $this->assertRegExp('#' . $pattern . '#', $query);
-    }
-
-    public function testWithCteAsExpression(): void
-    {
-        $cteQuery = $this->connection
-            ->newQuery()
-            ->select(['col' => 1]);
-
-        $query = $this->connection
-            ->newQuery()
-            ->with(new CommonTableExpression('cte', $cteQuery))
-            ->select('col')
-            ->from('cte');
-
-        $this->assertQuotedQuery(
-            'WITH cte AS \(SELECT 1 AS <col>\) SELECT <col> FROM <cte>',
-            $query->sql(),
-            !$this->autoQuote
-        );
-
-        $expected = [
-            [
-                'col' => '1',
-            ],
-        ];
-
-        $stmt = $query->execute();
-        $result = $stmt->fetchAll('assoc');
-        $stmt->closeCursor();
-        $this->assertEquals($expected, $result);
```

Comment:
We should keep a bunch of these integration tests. They will be helpful for ensuring the code works in the future and give us a reference point for the generated SQL and results that are created.

Reply:
As far as I know, I didn't discard any actual integration tests. I renamed the file to CommonTableExpressionQueryTests to match WindowQueryTests.

The only integration tests I removed were 2 that tested how sql server lets you use the CTE name in some place ... but they aren't relevant to how we generate any sql.

Reply:
:+1: It was hard to see that change in the diff output.